Model/HyvaPrefetchEventDispatcher.php

```diff
@@ -0,0 +1,54 @@
+<?php declare(strict_types=1);
+
+namespace Hyva\Admin\Model;
+
+use Magento\Framework\Event\ManagerInterface as EventManagerInterface;
+use Hyva\Admin\Model\GridSourceType\RepositorySourceType\HyvaAdminEventContainer;
+
+use function array_reduce as reduce;
+
+class HyvaPrefetchEventDispatcher
+{
+    private EventManagerInterface $eventManager;
+
+    public function __construct(EventManagerInterface $eventManager)
+    {
+        $this->eventManager = $eventManager;
+    }
+
+    public function dispatch(
+        string $gridName,
+        $data
+    ) {
+        return reduce(
+            [
+                'hyva_grid_source_prefetch_' . $this->getGridNameEventSuffix($gridName),
+            ],
+            fn (
+                $data,
+                string $eventName
+            ) => $this->dispatchEvent($gridName, $eventName, $data),
+            $data
+        );
```

Comment:
Because there is only a single event in the array, no reduce is needed.
Please replace it with
```
$eventName = 'hyva_grid_source_prefetch_' . $this->getGridNameEventSuffix($gridName);

return $this->dispatchEvent($gridName, $eventName, $data);
```